### PR TITLE
Improve NoCrystal

### PR DESCRIPTION
**Pull request by Old-Chum**
Added boolean value Sneak, which stops the module from placing while the player is sneaking.
Fixed bug where you could not switch to obsidian while the module could place.
Fixed bug where the module would not place supporting blocks.
Removed item entities from blocking placement. Stolen from [here](e5626e#diff-dbeee8794de38bcf5dfcb2b40c22195cR352-R357)

Let me know if I could improve this.

### CODE REVIEW

File: src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java [hunk 1]
```diff
@@ -59,10 +61,34 @@ public void onWalkingUpdate(EventUpdateWalkingPlayer event) {
             final BlockPos east = interpPos.east();
             final BlockPos west = interpPos.west();
 
-            final boolean canPlace = valid(north) || valid(south) || valid(east) || valid(west);
+            final BlockPos northBelow = interpPos.north().add(0, -1, 0);
+            final BlockPos southBelow = interpPos.south().add(0, -1, 0);
+            final BlockPos eastBelow = interpPos.east().add(0, -1, 0);
+            final BlockPos westBelow = interpPos.west().add(0, -1, 0);
```
Comment (cookiedragon234) [inline src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java]: use `.down()`

File: src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java [hunk 2]
```diff
@@ -236,6 +266,12 @@ private boolean valid(BlockPos pos) {
         final Block south = mc.world.getBlockState(pos.add(0, 0, 1)).getBlock();
         final Block east = mc.world.getBlockState(pos.add(1, 0, 0)).getBlock();
         final Block west = mc.world.getBlockState(pos.add(-1, 0, 0)).getBlock();
+        
+        for (Entity entity : mc.world.getEntitiesWithinAABBExcludingEntity(null, new AxisAlignedBB(pos))) {
+            if (entity instanceof EntityLivingBase) {
```
Comment (cookiedragon234) [inline src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java]: Some non living entities (such as falling blocks and minecarts) can block placing

File: src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java [hunk 3]
```diff
@@ -33,6 +32,8 @@
     private final Minecraft mc = Minecraft.getMinecraft();
 
     public final BooleanValue disable = new BooleanValue("Disable", new String[]{"dis"}, false);
+    // When true NoCrystal will not place while the player is sneaking
+    public final BooleanValue sneak = new BooleanValue("Sneak", new String[]{"sneak", "s"}, true);
```
Comment (cookiedragon234) [inline src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java]: Maybe change to noSneak? This is a confusing name.

File: src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java (hunk 2 above)
Comment (cookiedragon234) [reply src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java]: You can check if the entity has `preventEntitySpawning` set to true, or preferably use `mc.world.checkNoEntityCollision(bb, entity)`

**Comment by Old-Chum** [comment]
I implemented all the things you suggested and simplified and improved place() and valid().

**Pull request closed without merging**